py-polars/polars/dataframe/frame.py

```diff
@@ -4577,7 +4577,19 @@ def insert_column(self, index: int, column: Series) -> DataFrame:
         """
         if index < 0:
             index = len(self.columns) + index
```

Comment:
I just noticed that we get an `OverflowError`/`PanicException` if the index is out of bounds (negative or positive, respectively). Maybe throw in a tiny check to make sure the index is valid?

```python
df = pl.DataFrame({"a": [1]})
df.insert_column(-5, pl.Series("b", [2]))  # OverflowError: can't convert negative int to unsigned
df.insert_column(5, pl.Series("b", [2]))  # PanicException: insertion index (is 5) should be <= len (is 1)
```

Reply:
Good point; will update to handle that better ;)